internal/drivers/nomad/driver.go

```diff
@@ -423,13 +434,43 @@ func (p *config) fetchMachine(logr logger.Logger, id string) (ip, nodeID string,
 }
 
 // destroyJob returns a job targeted to the given node which stops and removes the VM
-func (p *config) destroyJob(vm, nodeID string, destroyGenerator func(string) string) (job *api.Job, id string) {
+func (p *config) destroyJob(vm, nodeID, storageIdentifier string, destroyGenerator func(string) string, storageCleanupType *storage.CleanupType) (job *api.Job, id string) {
 	id = destroyJobID(vm)
 	constraint := &api.Constraint{
 		LTarget: "${node.unique.id}",
 		RTarget: nodeID,
 		Operand: "=",
 	}
+	destroyCmd := destroyGenerator(vm)
+
+	var cephStorageScriptEncoded string
+	var cephStorageScriptPath string
+	if storageIdentifier != "" && storageCleanupType != nil {
```

Comment:
a separate function would be better, let's keep `destroyJob` easy to read. You can append the storage destroy command in this function